doc/contributing/hw/design.md

```diff
@@ -107,9 +107,24 @@ The sections below show how to do items 2-4.
 $ make -C hw top
 ```
 
-After you revise the IP `.hjson`, IP top module, IP register interface, and DV testbench `tb.sv`, you can re-generate top-levels with the command above.
+After you revise the IP `.hjson`, IP top module, IP register interface, and DV testbench `tb.sv`, you can re-generate top-levels and some top-dependent IPs with the commands above.
 This reads the `.hjson` file and updates the interrupt signal and re-generates the PLIC module if needed.
 
+### Update MD Files
+
+Some `*.md` files generate contents using the `util/cmdgen.py` tool described at [CMDGEN](../doc/README.md##cmdgen).
+These files are generated using cmdgen via make.
+
+```console
+$ make -C hw cmdgen
+```
+
+Generating both top and cmdgen can be done conveniently with
+
+```console
+$ make -C hw top_and_cmdgen
```

Comment:
How odd! Can't we just give `make -C hw top cmdgen` as the example (and get rid of the target in the Makefile)?

Reply:
That was my first attempt, but unfortunately the ci/scripts/check-generated.sh works by running a single target at a time but only running both these targets can we recreate all generated artifacts.

Reply:
Boo :-)